source/brailleDisplayDrivers/seikantk.py

```diff
@@ -290,6 +286,18 @@ def __init__(self, index):
 		self.routingIndex = index
 
 
+def _getKeyNames(keys: int) -> Set[int]:
+	return {_keyNames[1 << i] for i in range(16) if (1 << i) & keys}
+
+
+def _getDotNames(dots: int) -> Set[int]:
+	return {_dotNames[1 << i] for i in range(8) if (1 << i) & dots}
+
+
+def _getRoutingIndexes(routingKeys: bytes) -> Set[int]:
+	return {i * 8 + j for i in range(len(routingKeys)) for j in range(8) if routingKeys[i] & (1 << j)}
```

Comment:
This one is a little harder, I suspect it could be simplified. There are likely only so many values for `routingKeys`, I'd suggest a brute force unit test that tests them all (eg that the new algorithm gives the same out put as this one). Once simplified the test can probably be deleted.

Probably something like:
```suggestion
	routingKeyBitFlags = routingKeys
	bitValues = [1 << i for i in range(8)]
	routingIndexes = []
	routingIndexTest = 0
	for byte in routingKeyBitFlags:
		for bitValue in bitValues:
			if bitValue & byte:
				routingIdexes.append(routingIndexTest)
			routingIndexTest += 1
```
Or if you want to still do it in a list comprehension style
```suggestion
	bitsPerByte = 8
	bitValues = [1 << i for i in range(bitsPerByte)]
	return {
		bitIndex + (byteIndex * bitsPerByte)
		for bitIndex, bitValue in enumerate(bitValues)
		for byteIndex, bitFlags in enumerate(routingKeys)
		if bitValue & bitFlags
```

Reply:
I took a different approach here, closer to the approach suggested above. Let me know what you think.

The current testing should provide enough coverage here I think, unless Nippon releases a seika notetaker device with a very large number of keys.